FiveCalls/FiveCalls/Operator.swift

```diff
@@ -23,39 +23,43 @@ class Operator {
         queue = .main
     }
 
-    func fetchIssues(delegate: AppStateDelegate, completion: @escaping (OperationResult) -> Void) {
+    func fetchIssues(setIssues: @MainActor @escaping ([Issue]) -> ()) {
         let operation = FetchIssuesOperation()
         operation.completionBlock = { [weak operation] in
             if let issues = operation?.issuesList {
                 DispatchQueue.main.async {
-                    delegate.setIssues(issues: issues)
-                    completion(.success)
+                    setIssues(issues)
```

Comment:
In a unidirectional world of the type I'm used to this would be more like `appStore.send(.setIssues(issues))` where the store contains the current `appState` and has a reducer that handles the actions wrt the state.

Reply:
@nickoneill seem my rough approach here: https://github.com/5calls/ios/pull/338

Reply:
more of this approach is in this branch now